plugins/modules/rpm_repository.py

```diff
@@ -41,6 +41,10 @@
       - Max number of latest versions of each package to keep (optional)
     type: int
     version_added: "0.0.16"
+  retain_repo_versions:
+    description:
+      - Max number of repository versions to keep (optional)
```

Comment:
```suggestion
      - Max number of repository versions to keep
```
not being required is implied by ansible. Can you adjust line 41 too?

Also version_added may be nice.

Reply:
Thanks for the feedback, @mdellweg . I've made the adjustments and will submit them soon. 

Can you confirm what you would like to see regarding tests? I have produced 2 tests:

1. set `retain_repo_versions` to 5
2. update `retain_repo_versions` to 1 (i.e. don't keep old versions)

I have ran and recorded the tests--they all pass. However, I see the following now:

1. all existing `tests/fixtures/rpm_repository-*.yml` files have been modified
2. 2 new fixture files were created: `test/fixtures/rpm_repository-{22,23}.yml`

What portions of the tests/results would you like committed?

Side note:
There seems to be a limitation in which every call that includes the `retain_repo_versions` parameter updates the returned `pulp_last_updated` timestamp (no other returned values differ). This is the case even when `retain_repo_versions` is not modified. I assume this is a bug in `pulpcore` itself; however, it does limit the ability to assert `results.changed == false` on repeated calls with the same value for `retain_repo_versions`.

Reply:
> What portions of the tests/results would you like committed?

All of them. They are separate files for technical reasons only.

> Side note: [...]

That actually sounds like a bug on the client side. When we don't want to change the value, we must not do a patch call, and that still seems to happen.

Reply:
Hey @mdellweg, I have changed the tests to try for idempotence and am not seeing the previous issue. I believe that I must have made an error in my original work. Please see my changes and let me know if there is anything else needed for approval.

I appreciate you allowing me to contribute to this project.